src/main/java/io/github/novacrypto/bip32/PrivateKey.java

```diff
@@ -133,9 +133,14 @@ public PrivateKey cKDpriv(final int index) {
         final byte[] Ir = tail32(I);
 
         final byte[] key = hdKey.getKey();
-        final BigInteger mod = parse256(Il).add(parse256(key)).mod(n());
+        final BigInteger parse256_Il = parse256(Il);
+        final BigInteger ki = parse256_Il.add(parse256(key)).mod(n());
 
-        ser256(Il, mod);
+        if (parse256_Il.compareTo(n()) >= 0 || ki.equals(BigInteger.ZERO)) {
+            return cKDpriv(index + 1);
+        }
```

Comment:
In case parse256(IL) ≥ n or ki = 0, the resulting key is invalid, and one should proceed with the next value for i. (Note: this has probability lower than 1 in 2127.)